Assets/BossRoom/Scripts/Client/Net/ClientGameNetPortal.cs

```diff
@@ -103,14 +103,28 @@ private void OnConnectFinished(ConnectStatus status)
             //on failure, we must raise an event so that the UI layer can display something.
             Debug.Log("RecvConnectFinished Got status: " + status);
 
+            if( status != ConnectStatus.Success )
+            {
+                //this indicates a game level failure, rather than a network failure. See note in ServerGameNetPortal.
+                DisconnectReason.SetDisconnectReason(status);
+            }
+
             ConnectFinished?.Invoke(status);
         }
 
+        private void OnDisconnectReasonReceived(ConnectStatus status)
+        {
+            DisconnectReason.SetDisconnectReason(status);
+        }
+
         private void OnDisconnectOrTimeout(ulong clientID)
         {
             // we could also check whether the disconnect was us or the host, but the "interesting" question is whether
             //following the disconnect, we're no longer a Connected Client, so we just explicitly check that scenario.
-            if ( !MLAPI.NetworkManager.Singleton.IsConnectedClient && !MLAPI.NetworkManager.Singleton.IsHost )
+            bool disconnectedClient = !MLAPI.NetworkManager.Singleton.IsConnectedClient && !MLAPI.NetworkManager.Singleton.IsHost;

```

Comment:
waiting on your confirmation this works with hosts and "pulling out the cable"

Reply:
well, it didn't, so I guess that was worth testing. As this change didn't address any fixmes in particular, I've reverted it for now.